docs/our-approach-to-data.md

```diff
@@ -1,5 +1,7 @@
 # Our Approach to Data
 
+**Important notice:**: This document is slightly out of date. There's actually a fifth era of data fetching and manipulation in Calypso, using `@tanstack/react-query`, which we recommend for most of your data needs. We discourage the use of our legacy data layer (found in `client/state/data-layer`) – while there's definitely a case for using Redux as a state library and thunks for some data fetching, we generally recommend using `@tanstack/react-query` for most of your data needs. Please refer [`client/data/README.md`](https://github.com/Automattic/wp-calypso/blob/trunk/client/data/README.md) to get more information about usage conventions and recommendations (the document is currently a work in progress).
```

Comment:
I don't think this adds much to what we already have. The part that documents `client/data` is definitely useful, but this one can only add some confusion. I think we should update this document to give explanation and examples as to when to go with which approach. Also, the fact that we have a "fifth era" doesn't mean we're still not using and not recommending usage of data approaches from "previous eras". That being said, I'd love to have a thorough update here that can actually help developers instead of add additional confusion. WDYT?

Reply:
I think this document provides a lot of useful information on how to use the previous "eras" of data fetching in Calypso, especially the Redux related part (aka the 4th "era"), which is why I wanted to preserve what is stated there as that's still valuable – as you mentioned. I think not mentioning `@tanstack/react-query` in this document is even more confusing because that's what we actually recommend in most cases, no? We usually don't want people to use the legacy data layer – at least that's what I think. You're right about the part related to Redux thunks and state management I think, these can still be useful in some (rather rare?) cases.

I'm all open for expanding this document iteratively, as a first step I want to have something noted here rather than not mentioning RQ at all. I can totally rework the wording to ensure it's less confusing. Would you mind elaborating what's adding to the confusion?

Reply:
Incrementally improving the docs sounds good, and I see where you're coming from 👍 I also agree splitting the document into a historical one and a practical current one makes a lot of sense. Thanks!